### PR TITLE
Add user profile access

### PR DESCRIPTION
**Pull request by Nek-**

### CODE REVIEW

File: back/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -14,13 +14,19 @@
 use Behat\Behat\Context\SnippetAcceptingContext;
 use Behat\Gherkin\Node\PyStringNode;
 use Behat\Gherkin\Node\TableNode;
+use GuzzleHttp\Exception\BadResponseException;
 
 /**
  * Defines application features from the specific context.
  */
-class FeatureContext implements Context, SnippetAcceptingContext
+class FeatureContext extends \Knp\FriendlyContexts\Context\Context implements Context, SnippetAcceptingContext
```
Comment (valanz) [inline back/features/bootstrap/FeatureContext.php]: What about `use \Knp\FriendlyContexts\Context\Context as FriendlyContext`?

File: back/app/config/routing/users.yml [hunk 2]
```diff
@@ -1 +1,10 @@
 
+current_user_profile:
+    path: /profile.json
+    defaults: { _controller: 'CiTron:User/Profile:getUser' }
+    methods: [GET]
+
+user_profile:
+    path: /profile/{slug}.json
+    defaults: { _controller: 'CiTron:User/Profile:getUser' }
```
Comment (valanz) [inline back/app/config/routing/users.yml]: Should be another action, imo

File: back/src/CiTron/Controller/User/ProfileController.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+class ProfileController extends Controller
+{
+    /**
+     * @param User|null $user
+     * @return mixed|string
+     */
+    public function getUserAction(User $user = null)
+    {
+        if (null === $user) {
```
Comment (valanz) [inline back/src/CiTron/Controller/User/ProfileController.php]: The problem here is that if you try to get the profile of a non-existent user, you'll get your user detail. That's not a great behavior.

**Comment by Nek-** [comment]
@valanz fixed because I totaly agree your point !

**Pull request merged**